ggml/src/ggml.c

```diff
@@ -3839,24 +3834,11 @@ struct ggml_context * ggml_init(struct ggml_init_params params) {
         is_first_call = false;
     }
 
-    // find non-used context in g_state
-    struct ggml_context * ctx = NULL;
-
-    for (int i = 0; i < GGML_MAX_CONTEXTS; i++) {
-        if (!g_state.contexts[i].used) {
-            g_state.contexts[i].used = true;
-            ctx = &g_state.contexts[i].context;
-
-            GGML_PRINT_DEBUG("%s: found unused context %d\n", __func__, i);
-            break;
-        }
-    }
+    ggml_critical_section_end();
 
+    struct ggml_context * ctx = GGML_ALIGNED_MALLOC(sizeof(struct ggml_context));
```

Comment:
`GGML_ALIGNED_MALLOC` is likely completely overkill here, I don't think there is a good reason to require more than the alignment of `malloc`. Additionally, `GGML_ALIGNED_MALLOC` uses `vm_allocate` now on macOS, and this may actually be much slower than `malloc`, since it is requesting pages from the OS.